frontend/src/pages/UserDashboard/UserDashboard.jsx

```diff
@@ -1,74 +1,66 @@
-import { useState, useEffect } from 'react';
-import { useNavigate } from 'react-router-dom';
-import {  fetchEnrolledCourses, fetchUserProfile, fetchNotEnrolledCourses } from '../../services/api';
-import styles from './UserDashboard.module.css';
+import { useState, useEffect } from "react";
+import { useNavigate, useLocation } from "react-router-dom";
+import Home from "./Home";
+import MyCourses from "./MyCourses";
+import Profile from "./Profile";
+import styles from "./UserDashboard.module.css";
 
 const UserDashboard = () => {
-  const [activeTab, setActiveTab] = useState('Home');
-  const [profileData, setProfileData] = useState(null);
-  const [coursesData, setCoursesData] = useState([]);
-  const [enrolledCoursesData, setEnrolledCoursesData] = useState([]);
-  const userId = localStorage.getItem('userId'); 
+  const location = useLocation();
+  const [activeTab, setActiveTab] = useState(
+    location.state?.activeTab || "Home"
+  );
+  const userId = localStorage.getItem("userId");
   const navigate = useNavigate();
 
-  useEffect(() => {
-    const fetchData = async () => {
-      try {
-        if (activeTab === 'Profile') {
-          if (userId) {
-            const profile = await fetchUserProfile(userId);
-            console.log(profileData);
-            setProfileData(profile);
-          }
-        } else if (activeTab === 'Home') {
-          if (userId) {  
-            const courses = await fetchNotEnrolledCourses(userId);
-            setCoursesData(courses);
-          }
-        } else if (activeTab === 'My Courses') {
-          if (userId) {            
-            const enrolledCourses = await fetchEnrolledCourses(userId);
-            setEnrolledCoursesData(enrolledCourses);
-          }
-        }
-      } catch (error) {
-        console.error('Error fetching data:', error);
-      }
-    };
-
-    fetchData();
-  }, [activeTab, userId]);
-
   const handleCourseClick = (course) => {
-    const isEnrolled = activeTab === 'My Courses'; 
-    navigate(`/course/${course.courseName}`, { state: { course, isEnrolled } });
+    const isEnrolled = activeTab === "My Courses";
```

Comment:
This is a good case for string literals - someone is going to mistype "My Courses" at some point as "My courses" and the app will break. Not blocking but something to be aware of.